api/api_public_transaction_pool.go

```diff
@@ -88,19 +88,27 @@ func (s *PublicTransactionPoolAPI) GetTransactionByBlockHashAndIndex(ctx context
 // GetRawTransactionByBlockNumberAndIndex returns the bytes of the transaction for the given block number and index.
 func (s *PublicTransactionPoolAPI) GetRawTransactionByBlockNumberAndIndex(ctx context.Context, blockNr rpc.BlockNumber, index hexutil.Uint) (hexutil.Bytes, error) {
 	block, err := s.b.BlockByNumber(ctx, blockNr)
-	if block != nil && err == nil {
-		return newRPCRawTransactionFromBlockIndex(block, uint64(index)), nil
+	if err != nil {
+		return nil, err
 	}
-	return nil, err
+	tx := newRPCRawTransactionFromBlockIndex(block, uint64(index))
+	if tx == nil {
+		return nil, fmt.Errorf("the transaction does not exist (BlockNum: %d, Index:%d)", blockNr, index)
```

Comment:
`GetRawTransactionByBlockNumberAndIndex` was supposed to return (nil, nil) for the block having zero transactions, but this change make it return an error. It may cause an expected error case for existing API users. 
Is there any reason for this change?

Reply:
@JayChoi1736 I agree with @aidan-kwon. 
Returning nil seems enough.

Reply:
i thought return error would be better because formal case return method handler crash error. 
but i think return nil would be better. 
resolved by [205ceae](https://github.com/klaytn/klaytn/pull/1874/commits/205ceaec7a982d541336adc5385fa3c97e845f68)